codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -323,7 +127,8 @@ public boolean contains(T name) {
 
     @Override
     public boolean contains(T name, T value) {
-        return contains(name, value, keyComparator, valueComparator);
+        List<T> values = getAll(name);
```

Comment:
`getAll(..)` requires object allocation if an entry corresponding to `name` exists.  Either a `singletonList(..)` or `unmodifiableList(..)` is allocated.  Consider avoiding using `getAll(..)` for this purpose.